devito/ir/cgen/printer.py

```diff
@@ -234,75 +319,77 @@ def _print_Float(self, expr):
 
     def _print_FieldFromComposite(self, expr):
-        return "%s.%s" % (expr.pointer, expr.field)
+        return f"{expr.pointer}.{expr.field}"
 
     def _print_ListInitializer(self, expr):
-        return "{%s}" % ', '.join([self._print(i) for i in expr.params])
+        return f"{{{', '.join(self._print(i) for i in expr.params)}}}"
 
     def _print_IndexedPointer(self, expr):
-        return "%s%s" % (expr.base, ''.join('[%s]' % self._print(i) for i in expr.index))
+        return f"{expr.base}{''.join(f'[{self._print(i)}]' for i in expr.index)}"
 
     def _print_IntDiv(self, expr):
         lhs = self._print(expr.lhs)
         if not expr.lhs.is_Atom:
-            lhs = '(%s)' % (lhs)
+            lhs = f"({lhs})"
         rhs = self._print(expr.rhs)
         PREC = precedence(expr)
-        return self.parenthesize("%s / %s" % (lhs, rhs), PREC)
+        return self.parenthesize(f"{lhs} / {rhs}", PREC)
 
     def _print_InlineIf(self, expr):
         cond = self._print(expr.cond)
         true_expr = self._print(expr.true_expr)
         false_expr = self._print(expr.false_expr)
         PREC = precedence(expr)
-        return self.parenthesize("(%s) ? %s : %s" % (cond, true_expr, false_expr), PREC)
+        return self.parenthesize(f"({cond}) ? {true_expr} : {false_expr}", PREC)
 
-    def _print_UnaryOp(self, expr):
-        if expr.base.is_Symbol:
-            return "%s%s" % (expr._op, self._print(expr.base))
-        else:
-            return "%s(%s)" % (expr._op, self._print(expr.base))
+    def _print_UnaryOp(self, expr, op=None, parenthesize=False):
+        op = op or expr._op
+        base = self._print(expr.base)
+        if not expr.base.is_Symbol or parenthesize:
+            base = f'({base})'
+        return f'{op}{base}'
 
-    def _print_ComponentAccess(self, expr):
-        return "%s.%s" % (self._print(expr.base), expr.sindex)
+    def _print_Cast(self, expr):
+        cast = f'({self._print(expr._C_ctype)}{self._print(expr.stars)})'
+        return self._print_UnaryOp(expr, op=cast)
 
-    def _print_TrigonometricFunction(self, expr):
-        func_name = str(expr.func)
-        if self.single_prec():
-            func_name = '%sf' % func_name
-        return '%s(%s)' % (func_name, self._print(*expr.args))
+    def _print_ComponentAccess(self, expr):
+        return f"{self._print(expr.base)}.{expr.sindex}"
 
     def _print_DefFunction(self, expr):
         arguments = [self._print(i) for i in expr.arguments]
         if expr.template:
-            template = '<%s>' % ','.join([str(i) for i in expr.template])
+            ctemplate = ','.join([str(i) for i in expr.template])
+            template = f'<{ctemplate}>'
         else:
             template = ''
-        return "%s%s(%s)" % (expr.name, template, ','.join(arguments))
+        args = ','.join(arguments)
+        return f"{expr.name}{template}({args})"
 
-    _print_MathFunction = _print_DefFunction
+    def _print_SizeOf(self, expr):
```

Comment:
kinda weird we need a special handler for this one

Reply:
We used to have plaine string input type for SizeOf but not need to print the right type based on the printer so need its own handler.